worker/storageprovisioner/storageprovisioner.go

```diff
@@ -531,3 +530,7 @@ type context struct {
 	// machine.
 	managedFilesystemSource storage.FilesystemSource
 }
+
+func (c *context) IsApplicationKind() bool {
```

Comment:
Does this need to be exported? 
If so, it needs a comment, if not make it private.
Also, I think there's another place we can use this helper, where we add the watchers